bitbybit/src/bitfield.rs

```diff
@@ -584,58 +600,140 @@ fn parse_field(base_data_size: usize, field: &&Field) -> FieldDefinition {
+                        ).to_compile_error());
                     }
                     if end >= base_data_size {
-                        panic!("bitfield!: In Range {}..={}, end is out of range, as the base type has {} bits", start, end, base_data_size);
+                        return Err(syn::Error::new_spanned(
+                            &range_span,
+                            format!("bitfield!: In Range {}..={}, end is out of range, as the base type has {} bits", start, end, base_data_size)
+                        ).to_compile_error());
                     }
                     if start == end {
-                        panic!("bitfield!: In Range {start}..={end}, start is equal to end. Use the syntax 'bit({start})' instead", start = start, end = end)
+                        return Err(syn::Error::new_spanned(
+                            &attr.meta.require_list().unwrap().tokens,
+                            format!("bitfield!: In Range {start}..={end}, start is equal to end. Use the syntax 'bit({start})' instead", start = start, end = end)
+                        ).to_compile_error());
                     }
-                    range = Some(Range { start, end: end + 1 });
+                    range = Some(Range {
+                        start,
+                        end: end + 1,
+                    });
                 } else {
                     let bit_index_string = arguments[0];
                     let bit_index = bit_index_string.parse::<usize>()
-                        .unwrap_or_else(|x| panic!("bitfield!: Expected valid bit index, e.g. bit(8) but '{}' is not a number: {}", bit_index_string, x));
+                        .map_err(|e| if arguments[0] == "" {
+                            syn::Error::new_spanned(
+                                &attr.meta.require_list().unwrap().tokens,
+                                "bitfield!: Expected valid integer, but found an empty string"
+                            ).to_compile_error()
+                        } else {
+                            syn::Error::new_spanned(
+                                &range_span,
+                                format!("bitfield!: Expected valid bit index, e.g. bit(8) but '{}' is not a number: {}", bit_index_string, e)
+                            ).to_compile_error()
+                        })?;
                     if bit_index >= base_data_size {
-                        panic!("bitfield!: Bit index {} is out of range", bit_index);
+                        return Err(syn::Error::new_spanned(
+                            &range_span,
+                            format!("bitfield!: Bit index {} is out of range", bit_index),
+                        )
+                        .to_compile_error());
                     }
-                    range = Some(Range { start: bit_index, end: bit_index + 1 });
+                    range = Some(Range {
```

Comment:
did you run this through some formatter other than rustfmt or through rustfmt with custom settings?

Reply:
Nevermind this!